sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SpanExceptionRecorder.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+
+@FunctionalInterface
+public interface SpanExceptionRecorder {
+  AttributesBuilder recordException(Throwable exception, AttributesBuilder attributesBuilder);
```

Comment:
I toyed with having the interface be responsible for adding the event to the Span but I thought that I'd start here with attributes first.